test/e2e/framework/pod/resource.go

```diff
@@ -538,6 +539,16 @@ func checkPodsCondition(c clientset.Interface, ns string, podNames []string, tim
 	return success
 }
 
+// RemovePodFinalizer removes the pod's finalizer
+func RemovePodFinalizer(f *framework.Framework, pod *v1.Pod, finalizer string) {
+	pod, err := f.ClientSet.CoreV1().Pods(pod.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
```

Comment:
I'm not sure why. Should we retry in case of conflicts or wait until the pod is gone? I think we don't need to wait for the pod to be deleted as GC deletes it at some point anyway. Most of other tests (not using finalizers) do not wait actively for the pod to be gone.

Reply:
I see, updated using `pod_client.Update` which does retries